### PR TITLE
Dbeaver/pro#3414 highlighting disappear on ctrl+s sometimes

### PR DESCRIPTION
**Pull request by E1izabeth**
Bugs fixed:
1. Press Space, then Backspaсe several times in the middle of the script. After a while, the script's highlighting disappears after the edited place.
2. Write a query with errors (wrong table name, for example). Press Space, then Ctrl+S several times in the script before that query. After a while, you may encounter the disappearing of highlighting and red wavy lines under the error place.

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLEditorSemanticAnnotationModelFactory.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import org.eclipse.ui.texteditor.IMarkerUpdater;
+import org.eclipse.ui.texteditor.ResourceMarkerAnnotationModel;
+
+// see org.eclipse.ui.texteditor.ResourceMarkerAnnotationModelFactory
+public class SQLEditorSemanticAnnotationModelFactory implements IAnnotationModelFactory {
+
+    @Override
+    public IAnnotationModel createAnnotationModel(IPath location) {
+        IFile file = FileBuffers.getWorkspaceFileAtLocation(location);
+        return (IAnnotationModel)(file != null ? new ResourceMarkerSemanticAnnotationModel(file) : new AnnotationModel());
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLEditorSemanticAnnotationModelFactory.java]: Why the cast?

**Review by ShadelessFox — APPROVED** [review]
Love the comment :^)

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLEditorSemanticAnnotationModelFactory.java (hunk 1 above)
Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLEditorSemanticAnnotationModelFactory.java]: I copied this line from eclipse implementation

**Comment by Matvey16** [comment]
Verified

**Pull request merged**